source/codal_target_hal.cpp

```diff
@@ -51,8 +51,23 @@ int target_random(int max)
     return codal::random(max);
 }
 
-void target_wait_us(unsigned long us) {
-    codal::system_timer_wait_us(us);
+void target_wait_us(unsigned long us)
+{
+    if (us == 0)
+        return;
+
+    if (us < 3)
+        us = 3;
+
+    uint32_t n = (uint32_t)(us - 3) * (CODAL_CPU_CLOCK_SPEED / 1000000) / 3;
+    __asm__ __volatile__(
```

Comment:
I think this might be problematic, depending on flash cache lines. Hopefully, both instructions will be in a single word.

Maybe a better approach would be to have something like:

```
void delay(int cycles) {
for (int i = 0; i < cycles; ++i) __asm__ __volatile__ ("nop");
}
```

and the calibrate how many cycles you have to pass using a timer (i.e., run it for 100k cycles and measure time)? I think that should be safe from any flash placement and in fact could go in codal-core.

Reply:
background on cache lines: https://github.com/adafruit/circuitpython/blob/master/ports/atmel-samd/common-hal/neopixel_write/__init__.c#L80